pod_test.go

```diff
@@ -63,23 +75,89 @@ func TestPodCollector(t *testing.T) {
+				# TYPE pod_container_status_ready gauge
+				pod_container_status_ready{container="container1",namespace="ns1",pod="pod1"} 1
+				pod_container_status_ready{container="container2",namespace="ns2",pod="pod2"} 1
+				pod_container_status_ready{container="container3",namespace="ns2",pod="pod2"} 0
+				# HELP pod_container_status_restarts The number of container restarts per container.
+				# TYPE pod_container_status_restarts counter
+				pod_container_status_restarts{container="container1",namespace="ns1",pod="pod1"} 0
+				pod_container_status_restarts{container="container2",namespace="ns2",pod="pod2"} 0
+				pod_container_status_restarts{container="container3",namespace="ns2",pod="pod2"} 1
+				# HELP pod_container_status_running Describes whether the container is currently in running state.
+				# TYPE pod_container_status_running gauge
+				pod_container_status_running{container="container1",namespace="ns1",pod="pod1"} 1
+				pod_container_status_running{container="container2",namespace="ns2",pod="pod2"} 0
+				pod_container_status_running{container="container3",namespace="ns2",pod="pod2"} 0
+				# HELP pod_container_status_terminated Describes whether the container is currently in terminated state.
+				# TYPE pod_container_status_terminated gauge
+				pod_container_status_terminated{container="container1",namespace="ns1",pod="pod1"} 0
+				pod_container_status_terminated{container="container2",namespace="ns2",pod="pod2"} 1
+				pod_container_status_terminated{container="container3",namespace="ns2",pod="pod2"} 0
+				# HELP pod_container_status_waiting Describes whether the container is currently in waiting state.
+				# TYPE pod_container_status_waiting gauge
+				pod_container_status_waiting{container="container1",namespace="ns1",pod="pod1"} 0
+				pod_container_status_waiting{container="container2",namespace="ns2",pod="pod2"} 0
+				pod_container_status_waiting{container="container3",namespace="ns2",pod="pod2"} 1
+				# HELP pod_info Information about pod.
+				# TYPE pod_info gauge
+				pod_info{host_ip="1.1.1.1",namespace="ns1",pod="pod1",pod_ip="1.2.3.4"} 1
+				pod_info{host_ip="1.1.1.1",namespace="ns2",pod="pod2",pod_ip="2.3.4.5"} 1
+				# HELP pod_status_phase The pods current phase.
+				# TYPE pod_status_phase gauge
+				pod_status_phase{namespace="ns1",phase="Running",pod="pod1"} 1
+				pod_status_phase{namespace="ns2",phase="Pending",pod="pod2"} 1
+				# HELP pod_status_ready Describes whether the pod is ready to serve requests.
+				# TYPE pod_status_ready gauge
+				pod_status_ready{condition="false",namespace="ns1",pod="pod1"} 0
+				pod_status_ready{condition="false",namespace="ns2",pod="pod2"} 1
+				pod_status_ready{condition="true",namespace="ns1",pod="pod1"} 1
+				pod_status_ready{condition="true",namespace="ns2",pod="pod2"} 0
+				pod_status_ready{condition="unknown",namespace="ns1",pod="pod1"} 0
+				pod_status_ready{condition="unknown",namespace="ns2",pod="pod2"} 0
```

Comment:
Did you see my addition in the node metrics that allows you to only test a specific set of metrics per test case? Might be advisable here as well so you can catch all cases around condition and phase without blowing up the test case.